frontend/src/components/header/SmallHeader.js

```diff
@@ -0,0 +1,88 @@
+import HeaderTypography from "../HeaderTypography";
+import LinkActionButton from "../LinkActionButton";
+import Grid from "@mui/material/Grid";
+
+const DialogTransition = React.forwardRef(function Transition(props, ref) {
+	return <Slide direction="down" ref={ref} {...props} >{props.children}</Slide>;
+});
+
+/**
+ * Header that is displayed on mobile devices.
+ * @param userIsAdmin true if current user is an admin
+ * @param user user user object of the currently logged in admin
+ */
+export default function SmallHeader({ userIsAdmin, user }) {
+	const [open, setOpen] = useState(false);
+
+	const toggleMenuDialog = () => setOpen(prevOpen => !prevOpen);
+
+	return <Toolbar style={{ overflowX: "none", display: "flex", justifyContent: "space-between", paddingRight: "6px", paddingLeft: "0px", paddingTop: "3px" }}>
+		<div style={{ display: "inherit", paddingLeft: "4px" }}>
+			<Link to="/" style={{ textDecoration: "none", display: "inherit", userSelect: "none" }}>
+				<img src={NftFullLogo} alt="Logo" style={{ height: "27.55px", marginRight: "7px" }} onDragStart={(e) => e.preventDefault()}/>
+			</Link>
+		</div>
+		<div style={{ display: "inherit", marginTop: "1px", fontFamily: "PT Sans !important" }}>
+			{ user ?
+				<>
+					<RoundedEdgesButton color="inherit" component={Link} to="/user/profile" style={{ width: "80px", fontSize: "12px", height: "36px", }}>Profile</RoundedEdgesButton>
+				</>
+				:
+				<>
+					<RoundedEdgesButton style={{ backgroundColor: "#009C19", width: "99px", height: "36px", fontSize: "12px" }} component={Link} to="/signup">Sign Up</RoundedEdgesButton>
+				</>
+			}
+			<MenuIcon style={{ marginTop: "7px", marginLeft: "20px", cursor: "pointer" }} onClick={toggleMenuDialog}/>
+		</div>
+		<MenuDialog userIsAdmin={userIsAdmin} user={user} open={open} toggleMenuDialog={toggleMenuDialog}/>
+	</Toolbar>;
+}
+
+
+function MenuDialog({ user, userIsAdmin, open, toggleMenuDialog }) {
+	return <Dialog open={open} TransitionComponent={DialogTransition} keepMounted fullScreen>
+		<div style={{ backgroundColor: backgroundColor, height: "100%", width: "100%" }} >
+			<DialogTitle textAlign="right" style={{ paddingRight: "18px", paddingTop: "18px" }}><CloseIcon onClick={toggleMenuDialog} style={{ color: textColor }}/></DialogTitle>
+			<DialogContent style={{ marginTop: "160px", textAlign: "center" }} >
+				{ user
+					?
+					<>
+						{userIsAdmin &&<Grid item><LinkActionButton to="/user/admin" onClick={toggleMenuDialog} style={{ backgroundColor: "transparent", width: "192px", height: "54px", fontSize: "17px", border: "1px solid #FFFFFF", marginBottom: "26px" }} component={RoundedEdgesButton}>Admin</LinkActionButton></Grid>}
+						<Grid item><LinkActionButton to="/user/profile" onClick={toggleMenuDialog} component={RoundedEdgesButton} style={{ backgroundColor: "transparent", width: "192px", height: "54px", fontSize: "17px", border: "1px solid #FFFFFF" }}>Profile</LinkActionButton></Grid>
```

Comment:
I noticed, the mockups don't include any design view for the logged-in case. I overlooked that unfortunately. But I think you made a reasonable choice.

Your code looks like this dialog part here has a sliding animation when it appears/disappears. Nice!